File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/detail/folder/FolderDetailEvent.kt
package dev.subfly.yabacore.state.detail.folder

import dev.subfly.yabacore.model.ui.BookmarkUiModel
import dev.subfly.yabacore.model.utils.BookmarkAppearance
import dev.subfly.yabacore.model.utils.CardImageSizing
import dev.subfly.yabacore.model.utils.SortOrderType
import dev.subfly.yabacore.model.utils.SortType

sealed interface FolderDetailEvent {
    data class OnInit(val folderId: String) : FolderDetailEvent
    data class OnChangeQuery(val query: String) : FolderDetailEvent
    data object OnToggleSelectionMode : FolderDetailEvent
    data class OnToggleBookmarkSelection(val bookmarkId: String) : FolderDetailEvent
    data object OnDeleteSelected : FolderDetailEvent
    data class OnDeleteBookmark(val bookmark: BookmarkUiModel) : FolderDetailEvent
    data class OnChangeSort(val sortType: SortType, val sortOrder: SortOrderType) : FolderDetailEvent
    data class OnChangeAppearance(
        val appearance: BookmarkAppearance,
        val cardImageSizing: CardImageSizing? = null
    ) : FolderDetailEvent
}

File: Darwin/YABACore/State/Detail/Folder/FolderDetailEvent.swift
//
//  FolderDetailEvent.swift
//  YABACore
//
//  Parity with Compose `FolderDetailEvent` + folder metadata mutations.
//

import Foundation

public enum FolderDetailEvent: Sendable {
    case onInit(folderId: String)
    case onChangeQuery(String)
    case onToggleSelectionMode
    case onToggleBookmarkSelection(bookmarkId: String)
    case onDeleteSelected(bookmarkIds: [String])
    case onDeleteBookmark(bookmarkId: String)
    case onChangeSort(sortType: SortType, sortOrder: SortOrderType)
    case onChangeAppearance(appearance: BookmarkAppearance, cardImageSizing: CardImageSizing?)

    case updateMetadata(folderId: String, label: String, description: String?, icon: String, colorRaw: Int)
    case move(folderId: String, newParentId: String?)
    case delete(folderId: String)
}
